snuba/cli/replacer.py

```diff
@@ -91,19 +99,26 @@ def replacer(
     setup_logging(log_level)
     setup_sentry()
 
-    dataset = get_dataset(dataset_name)
+    storage = get_writable_storage(storage_name)
+
+    # If dataset_name is provided, use the writable storage from that dataset.
+    # This can be removed once we are passing storage_name instead of
+    # dataset_name everywhere
+    if dataset_name:
+        dataset = get_dataset(dataset_name)
+        storage = dataset.get_writable_storage()
 
-    stream_loader = enforce_table_writer(dataset).get_stream_loader()
+    stream_loader = storage.get_table_writer().get_stream_loader()
     default_replacement_topic_spec = stream_loader.get_replacement_topic_spec()
     assert (
         default_replacement_topic_spec is not None
-    ), f"Dataset {dataset} does not have a replacement topic."
+    ), f"Storage {storage} does not have a replacement topic."
     replacements_topic = replacements_topic or default_replacement_topic_spec.topic_name
 
     metrics = MetricsWrapper(
         environment.metrics,
         "replacer",
-        tags={"group": consumer_group, "dataset": dataset_name},
+        tags={"group": consumer_group, "storage": storage_name},
```

Comment:
This is going to log the wrong tag if you pass dataset_name and not the storage_name

Reply:
Thanks, updated